### DISCUSSION TITLE
Keep getting a field does not exist or is not a relational field

### DISCUSSION THREAD

**Discussion by rtnolan** [RobertCraigie/prisma-client-py#755] [Q&A] [answered]
Hi there,  

I've been a user of the JS/TS version of Prisma for quite a while now, so it was a nice surprise to see there was actually a Python implementation also. I'm not sure if I'm doing something incorrectly but I cannot get things to work. I've got the following in my schema is:
```javascript
generator client {
  provider  = "prisma-client-py"
  interface = "asyncio"
  enable_experimental_decimal = true
  recursive_type_depth = -1
}

datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}

model strategy {
  id                     String                   @id
  name                   String
  createdOn              DateTime                 @default(now())
}
```

I'm using FastAPI and I've followed the example for the most part to get things set up. After generating the client and executing the following:
```python

from prisma.models import strategy

async def get_strategy_id(self, id):
  try:
    cs = await strategy.prisma().find_unique(
      where={"id": {"equals": id}}, include={"id": True}
    )
    return cs

  except Exception as e:
    print(e)
```
This results in:
```
 File "/Users/tomnolan/code/repos/smart/venv/lib/python3.9/site-packages/prisma/builder.py", line 1948, in create_children
    model=builder.get_relational_model(
  File "/Users/tomnolan/code/repos/smart/venv/lib/python3.9/site-packages/prisma/builder.py", line 1482, in get_relational_model
    raise UnknownRelationalFieldError(model=current_model.__name__, field=field)
prisma.errors.UnknownRelationalFieldError: Field: "id" either does not exist or is not a relational field on the strategy model
```

But when I look at the model that's imported I can see:
```python
class strategy(bases.Basestrategy):
    """Represents a strategy record"""

    id: _str
    name: _str
    createdOn: datetime.datetime
```

I'm not quite where it is I'm going wrong. Not sure if it's worth mentioning that I'm also not getting any autocomplete in VSCode even though I have `pylance` installed, I also have another 300+ models in the schema file. Any thoughts as to what might be happening? Any help would be greatly appreciated. 

```
python==3.9.15
prisma==0.8.2
fastapi==0.95.0
```

**Comment by rtnolan** [comment]
Just a quick update, I've trimmed my `schema.prisma` down to just having one model and the same code now works so I'm starting to think it may be related to the size of my schema file.

**Comment by rtnolan** [comment]
I believe my query was actually just incorrect, shouldn't have been using an `equals` with a string, things appear to be all good now. I still don't have type completion when my schema is as large as it is but queries are at least returning now!

**Comment by RobertCraigie** [comment, marked as answer]
@rtnolan you're getting that error because `id` is not a relational field. Only relational fields can be used with `include`, if you want to select fields see this documentation: https://prisma-client-py.readthedocs.io/en/stable/reference/selecting-fields/
